brd-android/trade/src/main/java/com/fabriik/trade/ui/features/swap/SwapInputViewModel.kt

```diff
@@ -702,28 +741,27 @@ class SwapInputViewModel(
         state.sendingNetworkFee == null || state.receivingNetworkFee == null ->
             SwapInputContract.ErrorMessage.NetworkIssues
         state.sourceCryptoBalance < state.sourceCryptoAmount ->
-            SwapInputContract.ErrorMessage.InsufficientFunds(state.sourceCryptoBalance, state.sourceCryptoCurrency)
+            SwapInputContract.ErrorMessage.InsufficientFunds(state.sourceCryptoBalance,
+                state.sourceCryptoCurrency)
         state.sourceCryptoBalance < state.sourceCryptoAmount + state.sendingNetworkFee.cryptoAmountIfIncludedOrZero() ->
             SwapInputContract.ErrorMessage.InsufficientFundsForFee
         state.sourceCryptoAmount < state.minCryptoAmount ->
-            SwapInputContract.ErrorMessage.MinSwapAmount(state.minCryptoAmount, state.sourceCryptoCurrency)
-        state.sourceCryptoAmount > state.maxCryptoAmount ->
-            SwapInputContract.ErrorMessage.MaxSwapAmount(state.maxCryptoAmount, state.sourceCryptoCurrency)
+            SwapInputContract.ErrorMessage.MinSwapAmount(state.minCryptoAmount,
+                state.sourceCryptoCurrency)
+        state.sourceFiatAmount > state.dailySwapAmountLeft ->
+            if (state.isKyc1) SwapInputContract.ErrorMessage.Kyc1DailyLimit else SwapInputContract.ErrorMessage.Kyc2DailyLimit
+        state.isKyc1 && state.sourceFiatAmount > state.lifetimeSwapAmountLeft ->
+            SwapInputContract.ErrorMessage.Kyc1LifetimeLimit
         else -> null
     }
 
     private fun SwapInputContract.State.Loaded.validateAmounts() = copy(
-        swapErrorMessage = null,
         confirmButtonEnabled = !sourceCryptoAmount.isZero() && !destinationCryptoAmount.isZero()
     )
 
     private fun showSwapError(error: SwapInputContract.ErrorMessage) {
-        val state = currentLoadedState ?: return
-        setState {
-            state.copy(
-                swapErrorMessage = error,
-                confirmButtonEnabled = false
-            )
+        setEffect {
```

Comment:
```suggestion
        val state = currentLoadedState ?: return
        setState { state.copy(confirmButtonEnabled = false) }
        setEffect {
```

Reply:
Designs were changed so button is disabled only when no input is present and not when error is shown.